README.md

```diff
@@ -1,31 +1,148 @@
 
 ## Getting Started
 
-To get started, clone the repository and install the dependencies:
+### Installation
+
+To get started, clone the repository:
 
 ```bash
-git clone https://github.com/FilecoinFoundationWeb/filecoin-foundation
+git clone https://github.com/FilecoinFoundationWeb/filecoin-foundation.git
 cd filecoin-foundation
+```
+
+Then, install the dependencies:
+
+```bash
 npm install
 ```
 
-## Development
+At this point you should be able to run the development server for any of the applications.
+
+### Development
 
-To start the development server, run `npm run dev`, which is an alias for `turbo dev`. This command will start the development server for all applications and packages in the monorepo.
+Running `npx turbo dev` from the root of the monorepo will call the `dev` script for all applications and packages in the monorepo, as defined in each `package.json`. In our case, this will start the development server for all applications in our monorepo, packages do not have a `dev` script.
+
+In 99% of cases, you only want to start the development server for one application at a time. To so start do so, you should run `turbo dev` with a filter flag.
+
+For instance, to start the development server of the `ffdweb-site` application, run:
+
+```bash
+npx turbo dev --filter=ffdweb-site
+```
+
+You can also use the shorthand syntax:
+
+```bash
+npx turbo ffdweb-site#dev
+```
+
+> [!NOTE]
+> When using npm commands, such as `npm install`, the `--workspace` flag refers to the path of the application or package, relative to the root of the monorepo. When using turbo commands, such as `turbo dev`, the `--filter` flag refers to the name of the application or package, as defined in each `package.json` file.
+
+### Adding npm dependencies
+
+The `npm install <dependency>` command will install dependencies for all applications and packages in the monorepo. Generally, you want to add a dependency inside a specific application or package. To do so, run:
+
+```bash
+npm install <dependency> --workspace=<application/package>
+```
+
+> [!NOTE]
+> The install commands should be run from the root of the monorepo. Do not install dependencies inside the application or package directory, otherwise Turborepo will not be not aware of it and will not resolve the dependencies properly.
+
+For example, to add `jest` to the `ff-site` application, run:
+
+```bash
+npm install jest --workspace 'apps/ff-site'
+```
+
+Or the shorthand syntax:
+
+```bash
+npm install jest -w 'apps/ff-site'
+```
+
+This will update both the `apps/ff-site/package.json` and the root `package-lock.json` file.
+
+We also use [syncpack](https://www.npmjs.com/package/syncpack) to occasionally check and fix version mismatch across the monorepo with the following commands:
```

Comment:
`npm run check:versions` is actually included in our Husky pre-push hook, so version checks run automatically before every push. I do still like for us to mention the commands here explicitly so people know how to run them manually.